Nitric.Sdk/Secret/SecretVersion.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021, Nitric Technologies Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+using System;
+using Nitric.Proto.Secret.v1;
+using System.Text;
+namespace Nitric.Api.Secret
+{
+    public class SecretVersion
+    {
+        public readonly Secret Secret;
+        public readonly string Version;
+
+        public SecretVersion(Secret secret, string version)
+        {
+            if (secret == null || string.IsNullOrEmpty(secret.Name))
+            {
+                throw new ArgumentNullException(secret.Name);
+            }
+            if (string.IsNullOrEmpty(version))
+            {
+                throw new ArgumentNullException(version);
+            }
+            this.Secret = secret;
+            this.Version = version;
+        }
+        public byte[] Access()
```

Comment:
Chatting with Tim and have been moving to a new SDK pattern for this where the SecretVersion.Assess() method returns a new SecretValue class which contains a SecretVersion and the actual secret value data. The secret version is obtained from the RPC call, and is useful for data encryption use cases where we want to store the secret name and version against the encrypted data record so we can retrieve the secret value in the future to decrypt the record.

Please see the Java SDK or Node SDK branches for an examples.

Reply:
That has been updated to now return a SecretValue object